tests/integration/security/filebased_tls_origination/main_test.go

```diff
@@ -35,16 +36,24 @@ func TestMain(m *testing.M) {
 		NewSuite(m).
 		Label(label.CustomSetup).
 		Label("CustomSetup").
-		SkipExternalControlPlaneTopology().
 		Setup(istio.Setup(&inst, setupConfig, cert.CreateCustomEgressSecret)).
 		Run()
 }
 
-func setupConfig(_ resource.Context, cfg *istio.Config) {
+func setupConfig(ctx resource.Context, cfg *istio.Config) {
 	if cfg == nil {
 		return
 	}
-	cfg.ControlPlaneValues = `
+
+	var isExternalControlPlane bool
+	for _, cluster := range ctx.AllClusters() {
+		if cluster.IsExternalControlPlane() {
+			isExternalControlPlane = true
+		}
+	}
```

Comment:
Maybe it would be good to add an `IsExternalControlPlane()` method to `resource.Environment` and then this test, and the one from your previous PR, could both just call it.

```suggestion
	isExternalControlPlane := ctx.Environment().IsExternalControlPlane()
```